transparency/transparency.go

```diff
@@ -0,0 +1,163 @@
+// Copyright (c) The go-boot authors. All Rights Reserved.
+//
+// Use of this source code is governed by the license
+// that can be found in the LICENSE file.
+
+package transparency
+
+import (
+	"fmt"
+	"io/fs"
+
+	"github.com/usbarmory/go-boot/uefi/x64"
+	"github.com/usbarmory/boot-transparency/engine/sigsum"
+	"github.com/usbarmory/boot-transparency/policy"
+	"github.com/usbarmory/boot-transparency/transparency"
+)
+
+// Boot transparency configuration.
+var Config struct {
+	Status string
+
+	bootPolicy    []byte
+	witnessPolicy []byte
+	proofBundle   []byte
+	submitKey     []byte
+	logKey        []byte
+}
+
+const (
+	bootPolicyPath    = `\transparency\policy.json`
+	witnessPolicyPath = `\transparency\trust_policy`
+	proofBundlePath   = `\transparency\proof-bundle.json`
+	submitKeyPath     = `\transparency\submit-key.pub`
+	logKeyPath        = `\transparency\log-key.pub`
+)
+
+func init() {
+	Config.Status = "none"
+}
+
+// Load boot-transparency configuration from files on disk.
+func LoadConfig() (err error) {
+	root, err := x64.UEFI.Root()
+
+	if err != nil {
+		return fmt.Errorf("could not open root volume, %v", err)
+	}
+
+	Config.bootPolicy, err = fs.ReadFile(root, bootPolicyPath)
+	if err != nil {
+		return fmt.Errorf("cannot read boot policy, %v", err)
+	}
+
+	Config.witnessPolicy, err = fs.ReadFile(root, witnessPolicyPath)
+	if err != nil {
+		return fmt.Errorf("cannot read witness policy, %v", err)
+	}
+
+	Config.submitKey, err = fs.ReadFile(root, submitKeyPath)
+	if err != nil {
+		return fmt.Errorf("cannot read log submitter key, %v", err)
+	}
+
+	Config.logKey, err = fs.ReadFile(root, logKeyPath)
+	if err != nil {
+		return fmt.Errorf("cannot read log key, %v", err)
+	}
+
+	Config.proofBundle, err = fs.ReadFile(root, proofBundlePath)
+	if err != nil {
+		return fmt.Errorf("cannot read proof bundle, %v", err)
+	}
+
+	return
+}
+
+// Cleanup boot-transparency configuration.
+func CleanupConfig() {
```

Comment:
Probably best to have `Enable`, `Disable` as `CleanupConfig` has sort of a vague meaning (one could interpret it as a sanity check rather than actual disabling).